.circleci/config.yml

```diff
@@ -21,57 +21,55 @@ jobs:
       path:
         type: string
         description: The path to the orb
-      publish-docker-image:
-        type: boolean
-        description: |
-          Some orbs actions don't require us to maintain an executor image; set this to false if the orb uses existing public images
-        default: true
     docker:
       - image: 361339499037.dkr.ecr.eu-west-1.amazonaws.com/pe-orbs:latest
     steps:
       - checkout
       - setup_remote_docker:
           version: 17.11.0-ce
-      - when:
-          condition: <<parameters.publish-docker-image>>
-          steps:
-            - run:
-                name: docker build << parameters.path >>
-                # language=bash
-                command: |
-                  docker login --username $DOCKERHUB_USERNAME --password $DOCKERHUB_PASSWORD
-
-                  readonly ORB="<< parameters.path >>"
-
-                  for path in $ORB/executor/Dockerfile*; do
-
-                    filename=$(basename $path)
-
-                    if [[ "$filename" == "Dockerfile" ]]; then
-                        TAG="latest"
-                        BUILD_TAG="$(date "+%d-%m-%Y")"
-                    else
-                        TAG="${filename#"Dockerfile-"}"
-                        BUILD_TAG="${TAG}_$(date "+%d-%m-%Y")"
-                    fi
-
-                    docker build --tag "ovotech/${ORB}:${BUILD_TAG}" \
-                    --label org.label-schema.vcs-ref="$CIRCLE_SHA1" \
-                    --label org.label-schema.vcs-url="$CIRCLE_REPOSITORY_URL" \
-                    --label org.label-schema.schema-version="1.0" \
-                    --file $path \
-                    "${ORB}/executor"
-
-                    docker push "ovotech/${ORB}:${BUILD_TAG}"
-
-                    DIGEST=$(docker image inspect --format="{{index .RepoDigests 0}}" "ovotech/${ORB}:${BUILD_TAG}")
-                    echo $DIGEST > "${ORB}_${TAG}_digest.txt"
-
-                    if [ "$CIRCLE_BRANCH" = "master" ]; then
-                        docker tag "ovotech/${ORB}:${BUILD_TAG}" "ovotech/${ORB}:${TAG}"
-                        docker push "ovotech/${ORB}:${TAG}"
-                    fi
-                  done
+      - run:
+          name: docker build << parameters.path >>
+          # language=bash
+          command: |
+            docker login --username $DOCKERHUB_USERNAME --password $DOCKERHUB_PASSWORD
+
+            readonly ORB="<< parameters.path >>"
+
+            for path in $ORB/executor/Dockerfile*; do
+
+              if [ -e "$path" ]; then
+                 echo "Has docker files - continuing"
+              else
+                 echo "Has no docker files - exiting step"
```

Comment:
is this else required? seems like the path would always exist coming from the for loop?

Reply:
No - it's looking to see if there are docker files - bash seems strange and will still come in here if there are no files, but in that case I think it has a null.

Basically you're looping over the result of the ls - so if there are no files the ls is null but it still gives a return for that.